### ISSUE TITLE
[Bug] Blutooth Scan fails when BLUETOOTH_SCAN and BLUETOOTH_CONNECT is not set - Android Android 12 (API level 31)

### ISSUE THREAD

**Issue by pasha-o** [xamarin/Essentials#1943]
### Description
Currently when you switch to android 12 - and you are scanning for BT, the app crashes
Need android.permission.BLUETOOTH_SCAN permission for android.content.AttributionSource@f9e3658f: GattService registerScanner

currently Essentials doesn't support checking for adding BT SCAN and Connect required for Androind 12
https://developer.android.com/guide/topics/connectivity/bluetooth/permissions
### Steps to Reproduce

1. 
2. 
3. 

### Basic Information

- Version with issue:
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - Android:

**Comment by pasha-o** [comment]
thanks @AndreaGobs   -  will do

**Comment by rogihee** [comment]
What I did is create a DependencyService in my shared project with only Android having implementation like @AndreaGobs has. I added a check for the SDK version - 31 needs the new permissions, to scan in older devices you still need the location permissions.

```
[assembly: Dependency(typeof(PermissionsService))]
namespace MyProjectAndroid
{
    public class PermissionsService : IPermissionsService
    {
        public async Task<bool> RequestBlePermissions()
        {
                PermissionStatus status;
                // New SDK 31 API for bluetooth permissions
                if (Android.OS.Build.VERSION.SdkInt >= BuildVersionCodes.S)
                {
                    status = await Permissions.CheckStatusAsync<BluetoothPermission>();
                    bool showAlert = Permissions.ShouldShowRationale<BluetoothPermission>();
                    if (showAlert || status == PermissionStatus.Unknown || status == PermissionStatus.Denied)
                    {
                         // show an alert explaining why we need the permissions
                    }
                    status = await Permissions.RequestAsync<BluetoothPermission>();
                }
                else
                {
                    // < 31 API, use location permission to search and connect to devices
                    status = await Permissions.CheckStatusAsync<Permissions.LocationWhenInUse>();
                    bool showAlert = Permissions.ShouldShowRationale<BluetoothPermission>();
                    if (showAlert || status == PermissionStatus.Unknown || status == PermissionStatus.Denied)
                    {
                         // show an alert explaining why we need the permissions
                    }
                    status = await Permissions.RequestAsync<Permissions.LocationWhenInUse>(); });
                }
                return true;
            }
        }
}
```

The iOS version does nothing:
```
[assembly: Dependency(typeof(PermissionsService))]
namespace MyProjectiOS
{
    public class PermissionsService : IPermissionsService
    {
        public Task<bool> RequestBlePermissions(IAlertDialogService alertService)
        {
            return Task.FromResult(true);
        }

        public Task<bool> RequestLocationPermissions(IAlertDialogService alertService)
        {
            return Task.FromResult(true);
        }
    }
}
```
And shared:
```
public interface IPermissionsService
    {
        Task<bool> RequestBlePermissions();
    }
```

**Comment by syedjamal** [comment]
Hi @AndreaGobs  @rogihee 

I have tried to add the above code in my `Xamarin Forms Android` project but I get below error

```
Manifest.Permission doesn't contain a definition for BluetoothScan 
Manifest.Permission doesn't contain a definition for BluetoothConnect 
```

`BuildVersionCodes doesn't contain a definition for 'S'`

I have set Android project Target version as 
`<uses-sdk android:minSdkVersion="26" android:targetSdkVersion="31" />`

Could you please help to resolve the error

**Comment by rogihee** [comment]
@syedjamal these are the BLE permissions I have in my manifest:

```	<uses-permission android:name="android.permission.BLUETOOTH" />
	<uses-permission android:name="android.permission.BLUETOOTH_ADMIN" />
	<!-- < 31 permissions for BLE -->
	<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" android:maxSdkVersion="30" />
	<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" android:maxSdkVersion="30" />
	<!-- new 31 permissions for BLE -->
	<uses-permission android:name="android.permission.BLUETOOTH_SCAN" android:usesPermissionFlags="neverForLocation" />
	<uses-permission android:name="android.permission.BLUETOOTH_CONNECT" />```

**Comment by syedjamal** [comment]
@rogihee 

I have added permission in the` Android Manifest` file as you mentioned. But I still get the errors. It is not resolved.
I don't see any update for `Xamarin Android` in my visual studio for MAC 2019

```
Xamarin.Android
Version: 12.0.0.3 (Visual Studio Enterprise)
Commit: xamarin-android/d16-11/f0e3c2d
Android SDK: /Users/jamibr/Library/Android/sdk
	Supported Android versions:
		7.1 (API level 25)
		8.0 (API level 26)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 32.0.0
SDK Build Tools Version: 31.0.0
```

Is there anything I need to upgrde for support Android 12

**Comment by penghouho** [comment]
@syedjamal After I set the Droid project to have Target Framework with Android 12.0 (S), everything is okay now, before this I changed also the AndroidManifest.xml but it is not enough.

Right Click Droid Project -> Options -> Build -> General -> Target Framework -> Android 12.0 (S)

<img width="961" alt="image" src="https://user-images.githubusercontent.com/69447583/154199098-a0ff2a01-c5ec-44b9-bf72-d13c742d6ecb.png">

**Comment by janusw** [comment]
We also need the runtime handling of Android 12 Bluetooth permissions for xamarin-bluetooth-le, see xabre/xamarin-bluetooth-le#568.

**Comment by andreasbrostencab** [comment]
Why are this Bluetooth permissions not yet included in the Xamarin.Essentials project?

**Comment by janusw** [comment]
> Why are this Bluetooth permissions not yet included in the Xamarin.Essentials project?

Probably because the development of Xamarin.Essentials has slowed down significantly since the advent of Maui (which btw also does not have this feature yet, but I have just opened an issue there).

**Comment by andreasbrostencab** [comment]
> > Why are this Bluetooth permissions not yet included in the Xamarin.Essentials project?
> 
> Probably because the development of Xamarin.Essentials has slowed down significantly since the advent of Maui (which btw also does not have this feature yet, but I have just opened an issue there).

True, but as long as Maui isn't mature enought to move larger apps into, I see a huge need from Microsoft to keep Xamarin.Essentials up to date with changes like this.

**Comment by janusw** [comment]
> True, but as long as Maui isn't mature enought to move larger apps into, I see a huge need from Microsoft to keep Xamarin.Essentials up to date with changes like this.

I fully agree, but still I wouldn't bet on any new permissions coming to Xamarin at this point (unless you contribute it yourself). Also, with .NET 7, it seems like Maui is finally getting a bit closer to the point where it's stable enough for larger apps (but that surely depends on the use case).

**Comment by janseris** [comment]
> > > Why are this Bluetooth permissions not yet included in the Xamarin.Essentials project?
> > 
> > 
> > Probably because the development of Xamarin.Essentials has slowed down significantly since the advent of Maui (which btw also does not have this feature yet, but I have just opened an issue there).
> 
> True, but as long as Maui isn't mature enought to move larger apps into, I see a huge need from Microsoft to keep Xamarin.Essentials up to date with changes like this.

The Xamarin dev team basically moved to MAUI

**Comment by enantiomer2000** [comment]
I have a bunch of hardware which is still not compatible with Maui so stuck with Xamarin Forms atm...

**Comment by janusw** [comment]
> I thought they were going to support the Xamarin platform and essentials until Maui is fully ready for enterprise projects.

Xamarin will be officially supported until May 2024, but it's basically in "maintenance mode", so I guess you can expect bugfixes and minor updates, but no major new features.

**Comment by janusw** [comment]
> Been dealing for days on trying to just scan for beacons with android. Have yet to be successful and I ran into this issue. I spend more time trying to configure maui and trying to get it to work than actually coding....

Check out [dotnet-bluetooth-le](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le). It has sample clients for [Xamarin](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/tree/master/Source/BLE.Client/BLE.Client.Droid) and [MAUI](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/tree/master/Source/BLE.Client/BLE.Client.Maui) and both can handle Android 12 permissions well (via some [custom extensions](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/blob/master/Source/BLE.Client/BLE.Client.Droid/Helpers/PlatformHelpers.cs)). With .NET8, MAUI will support the Android 12 Bluetooth permissions out of the box, but I'm pretty sure this will not be coming to Xamarin any more. If you still have a Xamarin app that requires this, follow the technique used in the sample app above.

**Comment by janseris** [comment]
> > Been dealing for days on trying to just scan for beacons with android. Have yet to be successful and I ran into this issue. I spend more time trying to configure maui and trying to get it to work than actually coding....
> 
> Check out [dotnet-bluetooth-le](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le). It has sample clients for [Xamarin](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/tree/master/Source/BLE.Client/BLE.Client.Droid) and [MAUI](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/tree/master/Source/BLE.Client/BLE.Client.Maui) and both can handle Android 12 permissions well (via some [custom extensions](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/blob/master/Source/BLE.Client/BLE.Client.Droid/Helpers/PlatformHelpers.cs)). With .NET8, MAUI will support the Android 12 Bluetooth permissions out of the box, but I'm pretty sure this will not be coming to Xamarin any more. If you still have a Xamarin app that requires this, follow the technique used in the sample app above.

Yeah this works perfectly and I've created successfully a proof of concept bluetooth beacon app thanks to your repository.

**Comment by BConnectSoftware** [comment]
We made the go/no-go decision to drop maui and go native.  Sure its two code bases, but the feature set in maui is just not there for what we needed for BLE.  And the promise for .NET8, with maui's track record, I would not hold my breath that it works 100%.  Its unfortunate but honestly we are back up to speed and CODING and not configuring...

**Comment by janusw** [comment]
> We made the go/no-go decision to drop maui and go native. Sure its two code bases, but the feature set in maui is just not there for what we needed for BLE.

Hm, I can see various good reasons for dropping MAUI, but honestly lack of BLE support is not one of them, since this is covered quite well by community projects like dotnet-bluetooth-le.

**Comment by BConnectSoftware** [comment]
> Hm, I can see various good reasons for dropping MAUI, but honestly lack of BLE support is not one of them, since this is covered quite well by community projects like dotnet-bluetooth-le.

You realize that sometimes for legal and security reasons, open source is not an option for certain projects?
We did look at dotnet-bluetooth-le, but could not get it to successfully monitor/range beacons with android.
